### PR TITLE
helix: switch back to release tarball

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by yu-re-ka — CHANGES_REQUESTED** [review]
Once it is cached by hydra, you should not need to download all the `source` derivations anymore. unless there are references on them?
I don't understand why this is supposed to be "broken".

https://termbin.com/t1gh

**Comment by zowoq** [comment]
> Once it is cached by hydra, you should not need to download all the `source` derivations anymore. unless there are references on them? I don't understand why this is supposed to be "broken".
> 
> https://termbin.com/t1gh

See https://github.com/NixOS/nixpkgs/pull/232621#issuecomment-1557309679

**Comment by yu-re-ka** [comment]
> > Once it is cached by hydra, you should not need to download all the `source` derivations anymore. unless there are references on them? I don't understand why this is supposed to be "broken".
> > https://termbin.com/t1gh
> 
> See [#232621 (comment)](https://github.com/NixOS/nixpkgs/pull/232621#issuecomment-1557309679)

I see. then the references behave differently on darwin. But #233580 should fix both.

File: pkgs/applications/editors/helix/default.nix [hunk 1]
```diff
@@ -1,33 +1,27 @@
...
-  cargoSha256 = "sha256-/LCtfyDAA2JuioBD/CDMv6OOxM0B9A3PpuVP/YY5oF0=";
+  src = fetchzip {
+    url = "https://github.com/helix-editor/helix/releases/download/${version}/helix-${version}-source.tar.xz";
+    hash = "sha256-3ZEToXwW569P7IFLqz6Un8rClnWrW5RiYKmRVFt7My8=";
+    stripRoot = false;
+  };
+
+  cargoHash = "sha256-/LCtfyDAA2JuioBD/CDMv6OOxM0B9A3PpuVP/YY5oF0=";
 
   nativeBuildInputs = [ installShellFiles makeWrapper ];
```
Comment (the-mikedavis) [inline pkgs/applications/editors/helix/default.nix:17]: `git` must be available during the build: we use the `git` binary to check the revisions of the grammar repositories and fetch them if they're not what we expect.

The grammar sources are checked out at the proper revisions in the source tarball so I believe the "No such file or directory" is about the git binary instead.

I haven't looked through the nixpkgs history but maybe `git` was available to buildRustPlatform builds in the past and was removed while nixpkgs handled the grammar fetching itself?

**Review by yu-re-ka — APPROVED** [review]

File: pkgs/applications/editors/helix/default.nix (hunk 1 above)
Comment (zowoq) [reply pkgs/applications/editors/helix/default.nix]: > I haven't looked through the nixpkgs history but maybe `git` was available to buildRustPlatform builds in the past and was removed while nixpkgs handled the grammar fetching itself?

Looks like git was available via cargoLock that we were using for tree-sitter. https://github.com/NixOS/nixpkgs/pull/232621/files#diff-ca5625244fa586a1ce711c4fffc9a3d07f382e065458ab3c7d588d2950409cf3L19

Thank you @the-mikedavis.

**Comment by yu-re-ka** [comment]
`git` was **not** previously available. this could be viewed as a regression, but eh, it works.

**Review by yu-re-ka — CHANGES_REQUESTED** [review]
Can you set the updateScript to nix-update-script?

**Comment by zowoq** [comment]
> Can you set the updateScript to nix-update-script?

It isn't needed.

**Pull request merged**